streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -202,6 +204,10 @@ private void recordSendError(final String topic, final Exception exception, fina
                 "indicating the task may be migrated out";
             sendException.set(new TaskMigratedException(errorMessage, exception));
         } else {
+            // TODO: KIP-572 handle `TimeoutException extends RetriableException`
```

Comment:
As above: This is an open question. Input is welcome. I would like to tackle it in a follow up PR.